### PR TITLE
upgrade AnomalyDetector to use typespec library 0.41.0

### PR DESCRIPTION
**Pull request by chunyu3**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

**Comment by lirenhe** [comment]
@chunyu3, could you help to fix this?

"Unknown word (typespec), please fix the error or add words to ./custom-words.txt"

**Comment by raych1** [comment]
@weshaggard , would you like to remove package.json in this PR or in latter PR as tools in pipeline don't support global package.json yet?

**Review by lirenhe — APPROVED** [review]

File: specification/cognitiveservices/AnomalyDetector/package.json [hunk 1]
```diff
@@ -4,16 +4,17 @@
...
-    "@cadl-lang/compiler": "latest",
-    "@cadl-lang/openapi": "latest",
-    "@cadl-lang/openapi3": "latest",
-    "@cadl-lang/prettier-plugin-cadl": "latest",
-    "@cadl-lang/rest": "latest",
+    "@azure-tools/typespec-apiview": "latest",
+    "@azure-tools/typespec-autorest": "latest",
+    "@azure-tools/typespec-azure-core": "latest",
+    "@azure-tools/typespec-client-generator-core": "latest",
+    "@azure-tools/typespec-python": "latest",
```
Comment (mikekistler) [inline specification/cognitiveservices/AnomalyDetector/package.json:11]: This package is not available (yet), so tsp compile fails.

**Comment by mikekistler** [comment]
Will the updated openapi.json be added to this PR (when the compile works)?

**Comment by weshaggard** [comment]
>@weshaggard , would you like to remove package.json in this PR or in latter PR as tools in pipeline don't support global package.json yet?

I would like to remove package.json but I don't think we are ready just yet. For this PR I would simply minimize the package.json to just what is needed for the specs PR (i.e. autorest and apiview) and then we will remove this once we get the common one in the root of the repo updated to typespec.

File: specification/cognitiveservices/AnomalyDetector/tspconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+emit: [
+  "@azure-tools/typespec-autorest",
+  "@azure-tools/typespec-python",
+  "@azure-tools/typespec-csharp",
+]
+options:
+  "@azure-tools/typespec-autorest":
+    "output-file": "openapi.json"
```
Comment (weshaggard) [inline specification/cognitiveservices/AnomalyDetector/tspconfig.yaml:8]: Is this the path we want? Shouldn't we have this output into the correct path automatically?

File: specification/cognitiveservices/AnomalyDetector/tspconfig.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    "examples-directory": ./examples
+  "@azure-tools/typespec-python":
+    "basic-setup-py": true
+    "package-version": 3.0.0b6
```
Comment (weshaggard) [inline specification/cognitiveservices/AnomalyDetector/tspconfig.yaml:12]: @catalinaperalta do you know if we actually need this version here?

File: specification/cognitiveservices/AnomalyDetector/package.json (hunk 1 above)
Comment (chunyu3) [reply specification/cognitiveservices/AnomalyDetector/package.json:11]: removed un-needed python emitter dependency.

File: specification/cognitiveservices/AnomalyDetector/tspconfig.yaml (hunk 2 above)
Comment (chunyu3) [reply specification/cognitiveservices/AnomalyDetector/tspconfig.yaml:8]: This PR is for typespec renaming. Ignore this now.

File: specification/cognitiveservices/AnomalyDetector/tspconfig.yaml (hunk 3 above)
Comment (chunyu3) [reply specification/cognitiveservices/AnomalyDetector/tspconfig.yaml:12]: This PR is for typespec renaming. Ignore this now. and this is specifical for python sdk package version naming, and will contact python sdk person to refine this if needed.

**Pull request merged**